packages/docusaurus-theme-search/src/theme/SearchBar/index.module.css

```diff
@@ -0,0 +1,703 @@
+    to right,
+    var(--markprompt-primaryHighlight),
+    var(--markprompt-secondaryHighlight)
+  );
+  animation-name: progress;
+  animation-duration: 2s;
+  animation-fill-mode: none;
+  animation-iteration-count: infinite;
+  transition-timing-function: cubic-bezier(0.14, 0, 0.16, 1);
+  transition: opacity 200ms ease;
+}
+[data-loading-state='preload'] .progress {
+  opacity: 1;
+}
+[data-loading-state]:not([data-loading-state='preload']) .progress {
+  opacity: 0;
+}
+
+.references {
+  position: 'relative';
+  padding-top: 0.5rem;
+  padding-bottom: 0.5rem;
+  background-color: var(--markprompt-muted);
+  border-top: 1px solid var(--markprompt-border);
+  font-size: 0.75rem;
+  transition: height 500ms ease;
+  transform: translateY(100%);
+  opacity: 0;
+  animation: popup 200ms ease-out forwards;
+  width: 100%;
+  box-sizing: border-box;
+}
+.references[data-loading-state='preload'] {
+  height: 50px;
+}
+.references[data-loading-state='streaming-answer'],
+.references[data-loading-state='done'] {
+  height: 95px;
+}
+.references[data-loading-state='indeterminate'] {
+  display: none;
+  height: 0;
+}
+
+.references p {
+  transition: opacity 500ms ease;
+  position: absolute;
+  top: 0.25rem;
+  left: 2rem;
+  right: 2rem;
+  font-weight: 600;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+}
+.references[data-loading-state='preload'] > p:first-of-type {
+  opacity: 1;
+}
+.references[data-loading-state='preload'] > p:last-of-type {
+  opacity: 0;
+}
+[data-loading-state]:not([data-loading-state='preload']) > p:first-of-type {
+  opacity: 0 !important;
+}
+[data-loading-state]:not([data-loading-state='preload']) > p:last-of-type {
+  opacity: 1;
+}
+
+.references ul {
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: nowrap;
+  align-items: center;
+  gap: 0.5rem;
+  list-style-type: none;
+  margin-top: 2.25rem;
+  padding-left: 2rem;
+  padding-bottom: 2rem;
+  overflow-x: auto;
+  min-width: 100%;
+  width: 0;
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+}
+
+.references ul::-webkit-scrollbar {
+  display: none;
+}
+
+.reference {
+  font-size: 0.875rem;
+  line-height: 1.5rem;
+  animation-name: slide-up;
+  animation-duration: 1s;
+  animation-fill-mode: both;
+  transition-timing-function: ease-in-out;
+}
+
+.reference a {
+  display: inline-block;
+  text-decoration: none;
+  padding: 0.125rem 0.5rem;
+  border: 1px solid var(--markprompt-border);
+  border-radius: 0.375rem;
+  color: var(--markprompt-primary);
+  font-weight: 500;
+  transition-property: opacity;
+  transition-duration: 200ms;
+  white-space: nowrap;
+}
+
+.reference a:hover {
+  opacity: 0.8;
+}
+
+@keyframes contentShow {
+  from {
+    opacity: 0;
+    transform: translate(-50%, -46%) scale(0.98);
+  }
+  50% {
+    transform: translate(-50%, -51%) scale(1.02);
+  }
+  to {
+    opacity: 1;
+    transform: translate(-50%, -50%) scale(1);
+  }
+}
+
+@keyframes overlayShow {
```

Comment:
I think `@keyframes` should either be camel-case or kebab-case, not both.

Reply:
Could we benefit from adding `stylelint` to our repo?

Reply:
I definitely like to use `stylelint` with this much CSS.